docs/proposals/05-etcd-operator-tasks.md

```diff
@@ -0,0 +1,369 @@
+
+##### Possible scenarios
+
+* Full snapshot taken via compaction job is better as compared to full snapshot taken directly from etcd cluster as full snapshot taken via snapshot compaction is taken after [defragmentation](https://etcd.io/docs/v3.2/op-guide/maintenance/#defragmentation) which makes it smaller (more compacted) in size.
+* Full snapshot triggered on etcd cluster might cause load on running etcd cluster but full snapshot taken by compaction job is developed separately with zero effect on the running cluster.
+
+##### Task Config
+
+We do not need any config for this task. When creating an instance of `EtcdOperatorTask` for this scenario, `.spec.config` will be set to nil (unset).
+
+##### Pre-Conditions
+
+* There should not be a `on-demand snapshot-compaction` task already running for the same etcd cluster.
+
+>Note: `on-demand snapshot-compaction` runs as a separate job in a seprate pod, hence it doesn't depends on health of etcd cluster members or any other conditions.
+
+#### Trigger on-demand full/delta snapshot
+
+`Etcd` custom resource provides an ability to set [FullSnapshotSchedule](https://github.com/gardener/etcd-druid/blob/master/api/v1alpha1/types_etcd.go#L158) which is currently defauled to run once in 24 hrs. [DeltaSnapshotPeriod](https://github.com/gardener/etcd-druid/blob/master/api/v1alpha1/types_etcd.go#L171) is also made configurable which defines the duration after which a delta snapshot will be taken.
+If operator does not wish to wait for the scheduled full/delta snapshot, he/she can trigger an on-demand(out-of-schedule) full/delta snapshot on etcd cluster which will be taken by `leading-backup-restore`.
+
+##### Possible scenarios
+
+* [Shoot hibernation](https://github.com/gardener/gardener/blob/master/docs/usage/shoot_hibernate.md): Every etcd cluster incurs an inherent cost of preserving the volumes even when the shoot is in hibernation state. However it is possible to save costs by invoking this task to take a full snapshot before deleting the volumes.
+* [Control Plane Migration](https://github.com/gardener/gardener/blob/master/docs/proposals/07-shoot-control-plane-migration.md): In [gardener](https://github.com/gardener/gardener) a seed cluster control plane can be moved to another seed cluster. To prevent data loss and faster restoration of the etcd cluster in the target seed, a full snapshot can be triggered for the etcd cluster in the source seed.
+* A on-demand full snapshot can be triggered if scheduled snapshot fails due to any reason.
+
+##### Task Config
+
+```go
+// SnapshotType can be full or delta snapshot
+SnapshotType string `json:"snapshotType"`
```

Comment:
Isn't this statement supposed to define a type? Aren't JSON tags for fields of a struct?
Shouldn't it be:
```suggestion
type SnapshotType string
```

Reply:
This has been fixed in one of the newer commits.